packages/cli/src/WorkflowExecuteAdditionalData.ts

```diff
@@ -741,28 +742,33 @@ function hookFunctionsSaveWorker(): IWorkflowExecuteHooks {
 	};
 }
 
+export function findWorkflowStart(nodes: INode[]) {
```

Comment:
Maybe add an extra argument with the execution mode, so we can actually check what type of workflow to look for?

Reply:
Ah yes, now I realized the wording I used is terrible.

What I meant is have the `executionMode` as an additional arguments so we decide whether or not to look for `n8n-nodes-base.executeWorkflowTrigger` vs `n8n-nodes-base.start`.

If the ticket description is outdated then please disregard.

Reply:
Thank you, refactored. Please note that `findWorkflowStart()` is only for subworkflows and CLI-started workflows, and in these two cases the order is always the same. The refactoring should make this clear now.